sentence_transformers/SentenceTransformer.py

```diff
@@ -38,17 +39,27 @@ class SentenceTransformer(nn.Sequential):
     :param modules: This parameter can be used to create custom SentenceTransformer models from scratch.
     :param device: Device (like 'cuda' / 'cpu') that should be used for computation. If None, checks if a GPU can be used.
     :param cache_folder: Path to store models. Can be also set by SENTENCE_TRANSFORMERS_HOME enviroment variable.
-    :param use_auth_token: HuggingFace authentication token to download private models.
+    :param token: HuggingFace authentication token to download private models.
     """
     def __init__(self, model_name_or_path: Optional[str] = None,
                  modules: Optional[Iterable[nn.Module]] = None,
                  device: Optional[str] = None,
                  cache_folder: Optional[str] = None,
-                 use_auth_token: Union[bool, str, None] = None
+                 token: Optional[Union[bool, str]] = None,
+                 use_auth_token: Optional[Union[bool, str]] = None,
```

Comment:
maybe `Optional` is not needed? a default `False` would be better

Reply:
`Optional` is needed whenever `None` is a valid value to pass, which it is. I've kept `str` because the `token`/`use_auth_token` is passed directly to `transformers` and `huggingface_hub`, who use `Optional[Union[str, bool]]` as the typing:

https://github.com/huggingface/transformers/blob/f1185a4a73a03d238afce1b40456588d22520dd2/src/transformers/modeling_utils.py#L2303

Note also that you can either pass:
* `str`: The token to use as HTTP bearer authorization for remote files.
* `True`: will use the token generated when running `huggingface-cli login` (stored in `~/.huggingface`).
* `False`: will not use any token.
* `None`: Same as `True` (so: not same as `False`)

Reply:
i see, thanks for the clarification!